src/components/table/react/TableCell.tsx

```diff
@@ -0,0 +1,226 @@
+     */
+    isSortable?: boolean;
+
+    /**
+     * The scope of the thead, possible values: col/row.
+     */
+    scope?: Scope;
+
+    /**
+     * The initial sort order, possible values: asc/desc (sortable thead only).
+     */
+    sortOrder?: Order;
+
+    /**
+     * The variant of the cell, possible values: body/head.
+     */
+    variant?: Variant;
+
+    /**
+     * The function to call when we click on an order button.
+     */
+    onHeaderClick?(): void;
+}
+type TableCellProps = ITableCellProps;
+
+/////////////////////////////
+
+/**
+ * Define the types of the default props.
+ */
+interface IDefaultPropsType extends Partial<TableCellProps> {
+    variant: Variant;
+}
+
+/////////////////////////////
+//                         //
+//    Public attributes    //
+//                         //
+/////////////////////////////
+
+/**
+ * The display name of the component.
+ *
+ * @type {string}
+ * @constant
+ * @readonly
+ */
+const COMPONENT_NAME: string = `${COMPONENT_PREFIX}TableCell`;
+
+/**
+ * The default class name and classes prefix for this component.
+ *
+ * @type {string}
+ * @constant
+ * @readonly
+ */
+const CLASSNAME: string = getRootClassName(COMPONENT_NAME);
+
+/**
+ * The default value of props.
+ *
+ * @type {IDefaultPropsType}
+ * @constant
+ * @readonly
+ */
+const DEFAULT_PROPS: IDefaultPropsType = {
+    onHeaderClick: noop,
+    variant: Variants.body,
+};
+
+/////////////////////////////
+//                         //
+//    Private functions    //
+//                         //
+/////////////////////////////
+
+/**
+ * Validate the component props and children.
+ * Also, sanitize, cleanup and format the children and return the processed ones.
+ *
+ * @param  {TableCellProps}  props The children and props of the component.
+ * @return {React.ReactNode} The processed children of the component.
+ */
+function _validate(props: TableCellProps): React.ReactNode {
+    return validateComponent(COMPONENT_NAME, {
+        props,
+    });
+}
+
+/////////////////////////////
+
+/**
+ * The TableCell component displays an HTML Table Header Cell.
+ *
+ * @return {React.ReactElement} The component.
+ */
+const TableCell: React.FC<TableCellProps> = ({
+    children,
+    className = '',
+    icon,
+    isSortable,
+    onHeaderClick = DEFAULT_PROPS.onHeaderClick,
+    sortOrder,
+    variant = DEFAULT_PROPS.variant,
+    ...props
+}: TableCellProps): React.ReactElement => {
+    const newChildren: React.ReactNode = _validate({ children, onHeaderClick, variant, ...props });
+
+    /**
+     * Handle click on the ordered thead.
+     */
+    const handleOnHeaderClick: () => void = useCallback(() => {
+        if (isFunction(onHeaderClick)) {
+            onHeaderClick();
+        }
+    }, [onHeaderClick]);
+
+    return (
+        <Fragment>
+            {variant === Variants.head && (
+                <th
+                    className={classNames(
+                        'lumx-table__cell lumx-table__cell--header',
```

Comment:
All classes should begin with `CLASSNAME` also, many refacto with using `handleBasicClasses`